tagstudio/src/core/logging.py

```diff
@@ -0,0 +1,20 @@
+import logging
+from contextvars import ContextVar
+
+_root_logger = logging.getLogger()
+_main_logger = _root_logger.getChild("tag_studio")
+
+_main_logger_context_var: ContextVar[logging.Logger] = ContextVar("main_logger")
+_main_logger_context_var.set(_main_logger)
+
```

Comment:
> I'm guessing this is a thread safety feature since normally `getLogger(__name__)` will return the same logger reference without needing a global or contextvar
> 
> Forgive me if this is a common practice, I just haven't run across ContextVars before

The primary reason as to why a simple `getLogger(__name__)` was not used in the `get_logger` function is that in my opinion having some kind of "main logger" is logical in this particular instance, due to how the `__name__` variable won't contain the "tag_studio" part as tag_studio is not a package in the first place.

I am not certain whether the usage of contextvars for this purpose is a common practice anywhere, as I don't have much experience with real codebases. I have used a contextvar here in order to have a common "main logger" inside the execution context. I assume, that I could have simply used a top-level variable instead of a contextvar in this particular instance, and it wouldn't make any difference. There is no concrete reason as to why a contextvar is better here, it simply introduces unnecessary complexity and additional overhead. I could imagine a situation where someone would want different "main logger" for some parts of their system or subsystem but this is definetely not the case here. Would you suggest that I should use the already defined `_main_logger` variable inside the `get_logger` function and to get rid of the contextvar?

Regarding the thread safety, as far as I have understood after briedfly skimming the documentation and doing a small test, each thread will have it's own `contextvars.Context`, so the contextvar will introduce issues if someone where to try to call `get_logger` from a different thread. On the other hand, `logging.Logger.getChild` should be thread-safe and return the same `logging.Logger` object, provided the name is the same.